server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -1143,6 +1191,22 @@ private void ensureNoShardData(final NodePath[] nodePaths) throws IOException {
         }
     }
 
+    /**
+     * Throws an exception if cache exists on a non-search node.
+     */
+    private void ensureNoCacheData(final NodePath cacheNodePath) throws IOException {
+        List<Path> cacheDataPaths = collectCacheDataPath(cacheNodePath);
+        if (cacheDataPaths.isEmpty() == false) {
+            final String message = String.format(
+                Locale.ROOT,
+                "node does not have the %s role but has shard data: %s. Use 'opensearch-node repurpose' tool to clean up",
```

Comment:
Is the "but has shard data" text a copy/paste error?

Reply:
It is technically `cached shard data` but I reframed it to be more obvious.